### PR TITLE
Feature: Disable Bolt for customer groups

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
This PR is to add feature that allow admin disable Bolt for customer groups.
Here is the configuration: 
image

Fixes: https://app.asana.com/0/1204419514580775/1205280893811465

#changelog Disable Bolt for customer groups

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 

- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my ticket link and provided a changelog message.

### CODE REVIEW

**Review by vitaliyreznikov — COMMENTED** [review]
Discussed some changes offline

File: Block/BlockTrait.php [hunk 1]
```diff
@@ -156,6 +158,15 @@ protected function isIPRestricted()
         return $this->configHelper->isIPRestricted();
     }
 
+    /**
+     * @param int $customerGroupId
+     * @return bool
+     */
+    public function isCustomerDisabled($customerGroupId)
+    {
+        return in_array($customerGroupId, $this->configHelper->getDisabledCustomerGroups());
```
Comment (vitaliyreznikov) [inline Block/BlockTrait.php:167]: Let's make this function return false if the featureswith is disabled

Comment (johnmccombs1) [reply Block/BlockTrait.php:167]: @vitaliyreznikov  I've updated the PR.  Could you please re-check?

File: Helper/FeatureSwitch/Definitions.php [hunk 2]
```diff
@@ -665,5 +670,11 @@ class Definitions
             self::DEFAULT_VAL_KEY => false,
             self::ROLLOUT_KEY     => 0
         ],
+        self::M2_DISABLE_BOLT_FOR_CUSTOMER_GROUP => [
+            self::NAME_KEY        => self::M2_DISABLE_BOLT_FOR_CUSTOMER_GROUP,
```
Comment (vitaliyreznikov) [inline Helper/FeatureSwitch/Definitions.php:674]: Let's rename to something like "M2_ALLOW_DISABLING_BOLT_FOR_CUSTOMER_GROUP'.
With the current name, readers may be under the impression that something will be immediately disabled, but actually the feature switch only enable this option

File: etc/adminhtml/system.xml [hunk 3]
```diff
@@ -171,6 +178,7 @@
                         <label>Replace Button Selectors</label>
                         <config_path>payment/boltpay/replace_selectors</config_path>
                     </field>
+
```
Comment (vitaliyreznikov) [inline etc/adminhtml/system.xml:181]: Please remove the blank like

**Pull request merged**